pkg/controller/constraint/constraint_controller.go

```diff
@@ -66,7 +66,7 @@ type Adder struct {
 	Tracker          *readiness.Tracker
 	GetPod           func(context.Context) (*corev1.Pod, error)
 	ProcessExcluder  *process.Excluder
-	AssumeDeleted    func(schema.GroupVersionKind) bool
+	IfWatching       func(schema.GroupVersionKind, func() error) (bool, error)
```

Comment:
Add comment on what this is for

Reply:
adding:

```golang
	// IfWatching allows the reconciler to only execute functions if a constraint
	// template is currently being watched. It is designed to be atomic to avoid
	// race conditions between the constraint controller and the constraint template
	// controller
```